### DISCUSSION TITLE
Three spin Interaction model

### DISCUSSION THREAD

**Discussion by 000sadaf** [QuSpin/QuSpin#572] [Q&A]
Hi,
I have a Hamiltonian which has three spin interactions ,**H = J sum over i =0 to N sigma_x[i-1] *sigma_z[i]*sigma_x[i+1]** 
I transformed it into spinless fermion formalism and started to write the code; two problems occurred.
First, after writing the Periodic boundary conditions fermion operator with index[i-1] for the very first value of i that is (i=0) gives me -1, which is out of the index. It should be equal to L like [i+1]=[1], and [i-1]=[L]. How can i overcome this?? I tried IF condition, but it is not working. 
the second problem is, How to implement periodic boundary conditions and anti-periodic boundary conditions together. 
  Is it possible to write the code for my hamiltonian or should i use another package?
please help me out here.
Here is my code-
from quspin.basis import spinless_fermion_basis_1d
from quspin.operators import hamiltonian
import numpy as np
L= 15    ## for odd fermion numbers
J_x=1
J_y=3
h=1.0

basis=spinless_fermion_basis_1d(L=L)
J_nn_right=[[(J_x+J_y),(i-1),(i+1)%L] for i in range(L)]
J_nn_left=[[-(J_x+J_y),(i-1),(i+1)%L] for i in range(L)]
J_mm_right=[[(J_x-J_y),(i-1),(i+1)%L] for i in range(L)]
J_mm_left=[[-(J_x-J_y),(i-1),(i+1)%L] for i in range(L)]
#h_nn=[[2*h,i] for i in range(L)]
h_mm=[[h,i] for i in range(L)]
print(J_nn_right)
static=[["+-",J_nn_right],["-+",J_nn_left],["++",J_mm_right],["--",J_mm_left],["z",h_mm]]
dynamic=[]
H=hamiltonian(static,dynamic,dtype=np.float64,basis=basis)

**Comment by 000sadaf** [comment]
It feels like it is impossible to make code work using Quspin for this hamiltonian.

**Reply by PFmix** [reply]
I am not sure I understand your problem. Is your Hamiltonian:

$$
H= J \sum_{i=1}^L \sigma_i^x \sigma_{i+1}^z \sigma_{i+2}^x
$$

with PBC?

If so, why do you want to use the spinless fermion basis? Where are the $J_x$ and $J_y$ in your original Hamiltonian? As far as I understand, it is in general slower than using spins. Generating the Hamiltonian is then quite standard and only has a single boundary condition to consider:
```
from quspin.operators import hamiltonian
from quspin.basis import spin_basis_1d
import numpy as np

L= 15 ### or any parity
J= 1.0

basis=spinless_fermion_basis_1d(L=L)
J_list=[[J,i,(i+1)%L,(i+2)%L] for i in range(L)]
static=[["xzx",J_list]]
H=hamiltonian(static,dynamic,dtype=np.float64,basis=basis)
```
